File: Projects/PUBLIC_SHOP-APP/backend/src/main/java/tech/ksergei/bookslibrary/entity/Review.java
package tech.ksergei.bookslibrary.entity;

import java.sql.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import org.hibernate.annotations.CreationTimestamp;

import lombok.Data;

@Entity
@Table(name = "review")
@Data
public class Review {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "user_email")
    private String userEmail;

    @Column(name = "date")
    @CreationTimestamp
    private Date date;

    @Column(name = "rating")
    private double rating;

    @Column(name = "product_id")
    private Long productId;

    @Column(name = "review_text")
    private String reviewDescription;
    
}

File: Projects/PUBLIC_SHOP-APP/client/src/models/Review.ts
class Review {
    id: number;
    userEmail: string;
    date: string;
    rating: number;
    productId: number;
    reviewDescription?: string;

    constructor(
        id: number,
        userEmail: string,
        date: string,
        rating: number,
        productId: number,
        reviewDescription: string) {
        this.id = id;
        this.userEmail = userEmail;
        this.date = date;
        this.rating = rating;
        this.productId = productId;
        this.reviewDescription = reviewDescription;
    }
}

export default Review;